### PR TITLE
[Clang-16][D131307] suppress -Wenum-constexpr-conversion

### PR DESCRIPTION
**Pull request by junliume**
https://reviews.llvm.org/D131307

It is also causing problems in MIOpen and MIOpen dependencies:
```
[2022-10-29T14:44:59.883Z] ./boost/mpl/aux_/integral_wrapper.hpp:73:31: error: integer value -1 is outside the valid range of values [0, 3] for this enumeration type [-Wenum-constexpr-conversion]
```

### CODE REVIEW

File: src/comgr.cpp [hunk 1]
```diff
@@ -876,7 +876,11 @@ void BuildHip(const std::string& name,
             action.Do(AMD_COMGR_ACTION_LINK_RELOCATABLE_TO_EXECUTABLE, relocatable, exe);
         }
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wenum-constexpr-conversion"
         constexpr auto INTENTIONALY_UNKNOWN = static_cast<amd_comgr_status_t>(0xffff);
```
Comment (junliume) [inline src/comgr.cpp]: @atamazov any other ways to fix these similar issues than to suppress them?

**Comment by atamazov** [comment]
@junliume Just in case, according to https://reviews.llvm.org/D131307, it should be possible to use `-Wno-error=enum-constexpr-conversion` for transition period even with clang-16

> any other ways to fix these similar issues than to suppress them?

I am going to provide a patch with a proper fix for COMGR. WRT other cases (if any) it depends on design. Basically, it should be always possible.

**Comment by atamazov** [comment]
@junliume Please apply this patch to your branch and see if this helps:

* Revert COMGR changes
* COMGR: proper fix for enum-constexpr-conversion (pass "unknown problem" flag explicitly)

[comgr-fix-enum-constexpr-conversion.diff.txt](https://github.com/ROCmSoftwarePlatform/MIOpen/files/9955768/comgr-fix-enum-constexpr-conversion.diff.txt)

**Comment by atamazov** [comment]
@junliume ...alternatively, you can merge into this branch from https://github.com/atamazov/MIOpen/tree/clang_16_fix_ata-temp-0

**Review by atamazov — APPROVED** [review]
LGTM!

**Review by averinevg — APPROVED** [review]
LGTM

**Pull request merged**